save-backend/src/test/kotlin/com/saveourtool/save/backend/controllers/RunExecutionControllerTest.kt

```diff
@@ -0,0 +1,197 @@
+                .setResponseCode(200)
+        )
+        val assertions = sequence {
+            yield(mockServerOrchestrator.takeRequest(60, TimeUnit.SECONDS))
+        }.onEach {
+            log.info("Request $it")
+        }
+
+        val originalExecution = executionRepository.findById(EXECUTION_ID).get()
+        val originalExecutionIds = executionRepository.findAll()
+            .map { it.requiredId() }
+            .toList()
+        val originalTestExecutionIds = testExecutionRepository.findAll()
+            .map { it.requiredId() }
+            .toList()
+        webClient.post()
+            .uri("/api/$v1/run/re-trigger?executionId=$EXECUTION_ID")
+            .exchange()
+            .expectStatus()
+            .isAccepted
+        Thread.sleep(30_000)  // Time for request to create required entities
+
+        assertions.forEach { Assertions.assertNotNull(it) }
+        val newExecutions = executionRepository.findAll().toList()
+        Assertions.assertEquals(originalExecutionIds.size + 1, newExecutions.size)
+        val newExecution = newExecutions.first { it.requiredId() !in originalExecutionIds }
+        Assertions.assertEquals(originalExecution.project, newExecution.project)
+        Assertions.assertEquals("admin", newExecution.user?.name)
+        Assertions.assertEquals(originalExecution.testSuiteIds, newExecution.testSuiteIds)
+        Assertions.assertEquals(originalExecution.allTests, newExecution.allTests)
+        Assertions.assertEquals(originalExecution.additionalFiles, newExecution.additionalFiles)
+        Assertions.assertEquals(originalExecution.sdk, newExecution.sdk)
+        Assertions.assertEquals(originalExecution.execCmd, newExecution.execCmd)
+        Assertions.assertEquals(originalExecution.batchSizeForAnalyzer, newExecution.batchSizeForAnalyzer)
+
+        val newTestExecutions = testExecutionRepository.findAll().toList()
+        Assertions.assertEquals(originalTestExecutionIds.size + 1, newTestExecutions.size)
+    }
+
+    companion object {
+        private val log: Logger = getLogger<RunExecutionControllerTest>()
+        private const val EXECUTION_ID = 6L
+        private const val PROJECT_ID = 1L
+
+        @JvmStatic
+        lateinit var mockServerOrchestrator: MockWebServer
+
+        @AfterEach
+        fun cleanup() {
+            mockServerOrchestrator.checkQueues()
+            mockServerOrchestrator.cleanup()
+        }
+
+        @AfterAll
+        fun tearDown() {
+            mockServerOrchestrator.shutdown()
+        }
+
+        @DynamicPropertySource
+        @JvmStatic
+        fun properties(registry: DynamicPropertyRegistry) {
+            mockServerOrchestrator = createMockWebServer()
+            mockServerOrchestrator.start()
+            registry.add("backend.orchestratorUrl") { "http://localhost:${mockServerOrchestrator.port}" }
+        }
+    }
+
+    @TestConfiguration
+    class AdditionalConfiguration {
+        @Bean fun meterRegistry() = SimpleMeterRegistry()
```

Comment:
For tests it can be initialized with a `CompositeMeterRegistry`, which will wrap an empty list of registries hence acting effectively no-op

Reply:
thanks, will change